### PR TITLE
[coverity] detect: base64 checks DecodeBase64 return

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://scan5.scan.coverity.com/reports.htm#v56593/p10018/fileInstanceId=216329534&defectInstanceId=52603107&mergedDefectId=1504668

Describe changes:
- detect: base64 checks DecodeBase64 return
 
Introduced by 5b2761977871a94d559a9ba3b026593bb1ffd68b

cc @inashivb

### CODE REVIEW

**Review by inashivb — APPROVED** [review]
LGTM. Thanks. :)

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on tlpr1_alerts_cmp.
  

Pipeline 7741

File: src/detect-base64-decode.c [hunk 1]
```diff
@@ -82,24 +77,15 @@ int DetectBase64DecodeDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s
...
-    PrintRawDataFp(stdout, payload, decode_len);
-#endif
-
     uint32_t consumed = 0, num_decoded = 0;
-    (void)DecodeBase64(det_ctx->base64_decoded, det_ctx->base64_decoded_len_max, payload,
-            decode_len, &consumed, &num_decoded, BASE64_MODE_RELAX);
+    Base64Ecode code = DecodeBase64(det_ctx->base64_decoded, det_ctx->base64_decoded_len_max,
+            payload, decode_len, &consumed, &num_decoded, BASE64_MODE_RELAX);
+    if (code == BASE64_ECODE_ERR) {
+        return 0;
```
Comment (victorjulien) [inline src/detect-base64-decode.c:84]: pretty sure this is incorrect. We want to inspect available data even if at some point there is a data error

anyway, I believe the coverity error is already resolved in master by the `(void)` cast

Comment (victorjulien) [reply src/detect-base64-decode.c:84]: Confirmed by rerunning coverity on master (last run was on a dev branch)

Comment (inashivb) [reply src/detect-base64-decode.c:84]: failed to notice the relaxed mode.

Comment (catenacyber) [reply src/detect-base64-decode.c:84]: Oh...

How come coverity run on a dev branch ? Do you trigger builds manually ?

Comment (victorjulien) [reply src/detect-base64-decode.c:84]: yes all builds are triggered manually by me

Comment (catenacyber) [reply src/detect-base64-decode.c:84]: Ok thanks for the info, will check the branch/commit next time ;-)

(I thought it was automatic on master branch)

**Pull request closed without merging**